js/base-chat.js

```diff
@@ -0,0 +1,892 @@
+		});
+	}
+
+	d20plus.chat.getSpeakingTo = () => {
+		const prev = $("#speakingto").val();
+		$("#speakingto").html((() => {
+			return d20plus.chat.availableAddressees().reduce((result, addressee) => {
+				const icon = addressee.id ? "🗣" : "⚑";
+				const option = `${icon} ${addressee.name}`;
+				const value = `value="${addressee.name}"`;
+				result += `<option ${value}>${option}</option>`;
+				return result;
+			}, `<option value="">All</option><option value="ttms">None</option>`);
+		})());
+		$("#speakingto").val(prev);
+	}
+
+	addConfigOptions(
+		"chat", {
+			"social": {
+				"name": "Enable chat social panel (requires restart)",
+				"default": true,
+				"_type": "boolean",
+				"_player": true,
+			},
+			"showPlayerConnects": {
+				"name": "Show player connects messages",
+				"default": true,
+				"_type": "boolean",
+			},
+			"commands": {
+				"name": "Additional text chat commands (/help for full list)",
+				"default": true,
+				"_type": "boolean",
+				"_player": true,
+			},
+			"highlightttms": {
+				"name": "Highlight text box when in TTMS mode",
+				"default": true,
+				"_type": "boolean",
+				"_player": true,
+			},
+		},
+	);
+
+	d20plus.chat.onSocial = () => {
+		const $inputContainer = $("#textchat-input");
+		if (!d20plus.chat.social) {
+			const resized = $inputContainer.attr("style").includes("height")
+			if (resized) $inputContainer.addClass("social-resized");
+			else $inputContainer.addClass("social-default");
+			d20plus.chat.refreshLanguages();
+			d20plus.chat.getSpeakingTo();
+			d20plus.chat.social = true;
+		} else {
+			d20plus.chat.closeSocial();
+		}
+	}
+
+	d20plus.chat.onSpeakingAs = () => {
+		d20plus.chat.refreshLanguages();
+	}
+
+	d20plus.chat.onSpeakingTo = () => {
+		const speakingTo = $("#speakingto").val();
+		const ttms = speakingTo === "ttms";
+		if (d20.textchat.talktomyself && !ttms) {
+			d20.textchat.doChatInput(`/talktomyself off`);
+			d20plus.chat.localHistory.push(false);
+			setTimeout(() => d20plus.chat.checkTTMSStatus(), 10);
+		} else if (!d20.textchat.talktomyself && ttms) {
+			d20.textchat.doChatInput(`/talktomyself on`);
+			d20plus.chat.localHistory.push(false);
+			setTimeout(() => d20plus.chat.checkTTMSStatus(), 10);
+		}
+		if (speakingTo && !ttms) {
+			$("#textchat-social-notifier").addClass("b20-to");
+			$("#textchat-social-notifier-to").text(speakingTo);
+		} else {
+			$("#textchat-social-notifier").removeClass("b20-to");
+		}
+	}
+
+	d20plus.chat.onSpeakingIn = () => {
+		const speakingIn = $("#speakingin").val();
+		if (speakingIn) {
+			$("#textchat-social-notifier").addClass("b20-in");
+			$("#textchat-social-notifier-in").text(speakingIn);
+		} else {
+			$("#textchat-social-notifier").removeClass("b20-in");
+		}
+	}
+
+	d20plus.chat.resetSocial = () => {
+		if (!d20.textchat.talktomyself) $("#speakingto").val("");
+		$("#speakingin").val("");
+		$("#textchat-social-notifier").removeClass("b20-in");
```

Comment:
extract `#textchat-social-notifier` selector and/or chain/array `.removeClass` (see below comment)

Reply:
same as above